dsl.go

```diff
@@ -498,7 +501,8 @@ func NewOperation() *Operation {
 
 // SetTags sets the Tags of the Operation.
 func (o *Operation) SetTags(ts []string) *Operation {
-	o.Tags = ts
+	o.Tags = make([]string, len(ts))
```

Comment:
I think we can use [`slices.Clone`](https://pkg.go.dev/golang.org/x/exp/slices#Clone).

Reply:
In go1.21 we can use this:
https://github.com/golang/go/blob/master/src/slices/slices.go#L162
https://github.com/golang/go/blob/master/src/maps/maps.go#L61

Reply:
We're already using `x/exp/slices` anyway.